python/cudf/cudf/dataframe/numerical.py

```diff
@@ -424,6 +424,9 @@ def numeric_column_binop(lhs, rhs, op, out_dtype, reflect=False):
     # Call and fix null_count
     null_count = cpp_binops.apply_op(lhs, rhs, out, op)
 
+    if op in ['lt', 'gt', 'le', 'ge', 'eq', 'ne']:
```

Comment:
Don't disable the mask. If you want, you can use here the logic to `fillna` based on operator used. But I'd suggest that it should go into `Series`. That way, the fix would also work for column types other than `Numerical`